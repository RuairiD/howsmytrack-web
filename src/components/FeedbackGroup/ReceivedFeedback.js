import React from "react";

import { Card, Empty, List } from "antd";

import FeedbackResponse from "../FeedbackResponse/FeedbackResponse";

const LIST_GRID_LAYOUT = {
    gutter: 16,
    xs: 1,
    sm: 1,
    md: 1,
    lg: 1,
    xl: 1,
    xxl: 2,
};

const ReceivedFeedback = ({ feedbackReceived }) => {
    if (feedbackReceived === null || feedbackReceived === undefined) {
        return (
            <Card>
                <Empty
                    style={{ margin: "1em" }}
                    image={Empty.PRESENTED_IMAGE_DEFAULT}
                    description="Feedback for your submission will appear here once you have submitted feedback for everyone else in your group."
                />
            </Card>
        );
    }

    if (feedbackReceived.length === 0) {
        return (
            <Card>
                <Empty
                    style={{ margin: "1em" }}
                    image={Empty.PRESENTED_IMAGE_SIMPLE}
                    description="Nobody in your group has submitted feedback yet..."
                />
            </Card>
        );
    }

    return (
        <List
            grid={LIST_GRID_LAYOUT}
            dataSource={feedbackReceived}
            renderItem={(feedbackItemReceived) => (
                <List.Item>
                    <FeedbackResponse
                        {...feedbackItemReceived}
                    />
                </List.Item>
            )}
        />
    );
};

export default ReceivedFeedback;
