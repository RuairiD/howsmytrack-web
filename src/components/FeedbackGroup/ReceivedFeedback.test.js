import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ReceivedFeedback from "./ReceivedFeedback";

describe("ReceivedFeedback", () => {
    it("renders an error state if feedback is null (implying user hasn't written their own feedback yet)", () => {
        const wrapper = shallow(
            <ReceivedFeedback
                feedbackReceived={null}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders an empty state if no feedback is provided", () => {
        const wrapper = shallow(
            <ReceivedFeedback
                feedbackReceived={[]}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders a list of feedback, if feedback is provided", () => {
        const wrapper = shallow(
            <ReceivedFeedback
                feedbackReceived={[
                    {
                        feedbackResponseId: 1901,
                        feedback: "pretty good",
                        currentRating: 5,
                        allowReplies: false,
                        replies: 3,
                        unreadReplies: 2,
                    },
                ]}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders individual FeedbackResponse items", () => {
        const wrapper = mount(
            <ReceivedFeedback
                feedbackReceived={[
                    {
                        feedbackResponseId: 1901,
                        feedback: "pretty good",
                        currentRating: 5,
                        allowReplies: false,
                        replies: 3,
                        unreadReplies: 2,
                    },
                ]}
            />,
        );

        expect(wrapper.find("FeedbackResponse").length).toBe(1);
    });
});
