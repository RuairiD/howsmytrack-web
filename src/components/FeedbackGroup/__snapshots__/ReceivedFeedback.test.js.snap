// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceivedFeedback renders a list of feedback, if feedback is provided 1`] = `
<List
  bordered={false}
  dataSource={
    Array [
      Object {
        "allowReplies": false,
        "currentRating": 5,
        "feedback": "pretty good",
        "feedbackResponseId": 1901,
        "replies": 3,
        "unreadReplies": 2,
      },
    ]
  }
  grid={
    Object {
      "gutter": 16,
      "lg": 1,
      "md": 1,
      "sm": 1,
      "xl": 1,
      "xs": 1,
      "xxl": 2,
    }
  }
  loading={false}
  pagination={false}
  renderItem={[Function]}
  split={true}
/>
`;

exports[`ReceivedFeedback renders an empty state if no feedback is provided 1`] = `
<Card>
  <Empty
    description="Nobody in your group has submitted feedback yet..."
    image={<Simple />}
    style={
      Object {
        "margin": "1em",
      }
    }
  />
</Card>
`;

exports[`ReceivedFeedback renders an error state if feedback is null (implying user hasn't written their own feedback yet) 1`] = `
<Card>
  <Empty
    description="Feedback for your submission will appear here once you have submitted feedback for everyone else in your group."
    image={<Empty />}
    style={
      Object {
        "margin": "1em",
      }
    }
  />
</Card>
`;
